### PR TITLE
[mihome] add new channel for smoke sensor & minor improvements

### PR DESCRIPTION
**Pull request by dimalo**
Signed-off-by: Dieter Schmidt <dima.loschkin@gmail.com>

[mihome] add new channel for smoke sensor & minor improvements

- New channel for smoke sensor! Credits to [the Community](https://community.openhab.org/t/support-for-more-mi-home-aqara-items/32122/46)
-> can be useful for air quality measurements too
- smoke & gas sensor now show status "OK" instead of "Release alarm"

### CODE REVIEW

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_smoke.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
 		</supported-bridge-type-refs>
 		<label>Xiaomi Smoke Sensor</label>
 		<channels>
+            <channel id="density" typeId="density"></channel>
 			<channel id="alarm" typeId="smoke_alarm"></channel>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_smoke.xml]: Formatting looks funny, please rember that the XML files should be formatted using tabs.

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorBaseAlarmHandler.java [hunk 2]
```diff
@@ -38,11 +38,13 @@ public XiaomiSensorBaseAlarmHandler(Thing thing) {
     void parseReport(JsonObject data) {
         if (data.has(ALARM)) {
             int alarm = data.get(ALARM).getAsInt();
-            if (alarm >= 1 && alarm <= 2) {
+            // alarm channel only receives the "real alarm"
+            if (alarm == 1) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorBaseAlarmHandler.java]: 2 does not play along anymore?

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_smoke.xml (hunk 1 above)
Comment (dimalo) [reply addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_smoke.xml]: ✅

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorBaseAlarmHandler.java (hunk 2 above)
Comment (dimalo) [reply addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorBaseAlarmHandler.java]: "fire alarm only" matches the channel category for ESH more
2 and the other status messages (see ALARM_STATUS_MAP for gas & smoke handler) relate more to hardware alarms which are covered by the status channel (which nevertheless shows "fire alarm" when alarm==1)
this way users can conveniently call rules when the real alarm goes off with the switch and monitor the device via the status String channel

**Comment by kubawolanin** [comment]
LGTM! :-)

**Comment by martinvw** [comment]
Thanks to all of you!

**Pull request merged**